File: 128.Longest_Consecutive_Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> nums_set(nums.begin(), nums.end());
        int max_length = 0;
        for(auto num: nums){
            if(nums_set.count(num - 1) == 0){
                int length = 1;
                while(nums_set.count(++num))
                    length++;
                max_length = max(max_length, length);
            }
        }
        return max_length;
    }
};

File: 128. Longest Consecutive Sequence/128. Longest Consecutive Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        //transfer vector to ordered set
        Set<Integer> nums_set = new HashSet<>();
        for(var num: nums){
            nums_set.add(num);
        }
        int max_length = 0;
        for(var num: nums){
            int length = 1;
            //make sure the num is the start of a consecutive sequence
            if(!nums_set.contains(num - 1)){   
                //if exist the next consecutive num
                while(nums_set.contains(num + 1)){
                    length++;
                    num++;
                }
                max_length = Math.max(max_length, length);
            }
        }
        return max_length;
    }
}
